### PR TITLE
add e2e tests

### PR DESCRIPTION
**Pull request by restrry**
basic implementation.if you are okay, I can switch tests from `vows` to `jest` as well and add more tests later.
Right now I want to have at least something so I can add new functionality/change existing one without breaking changes.
The comparison can be done with snapshot testing, but I'd prefer to keep files as plain `.ts` files, so we can open them in IDE and make sure they are correct.

### CODE REVIEW

File: __tests__/fixtures/protected/expected.ts [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+type Logger = {
+    log: (line: string) => any
+};
+
+/**
+ * 
+ * @class Apiprotected
+ * @param {(string)} [domainOrOptions] - The project domain.
+ */
+export default class Apiprotected {
```
Comment (mtennoe) [inline __tests__/fixtures/protected/expected.ts]: Nit: ApiProtected

File: __tests__/fixtures/users/expected.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+type Logger = {
+    log: (line: string) => any
+};
+
+/**
+ * 
+ * @class Apiusers
+ * @param {(string)} [domainOrOptions] - The project domain.
+ */
+export default class Apiusers {
```
Comment (mtennoe) [inline __tests__/fixtures/users/expected.ts]: nit: ApiUsers

**Comment by mtennoe** [comment]
Great work!
I personally prefer jest :)
This reminds me that I need to get started the work of rewriting the whole thing to typescript as well.

**Comment by restrry** [comment]
@mtennoe addressed

**Comment by mtennoe** [comment]
@restrry Thanks! Merged!

**Pull request merged**